app/controllers/overrides/devise_token_auth_overrides/registrations_controller.rb

```diff
@@ -5,6 +5,12 @@ class RegistrationsController < DeviseTokenAuth::RegistrationsController
       before_action :permit_extra_params, only: %i[create update]
       skip_before_action :validate_account_update_params, only: :update
 
+      def resource_data(_opts = {})
+        response_data = UserSerializer.new(@resource).as_json['data']
+        response_data['type'] = @resource.class.name.parameterize if json_api?
```

Comment:
This line of code is pointless for us, because json_api? is implemented weirdly in the gem. It just checks if ActiveModelSerializer is being used or not, which will never be the case for us since we use fast_jsonapi for serialization instead of AMS.